lib/AST/ExprConstant.cpp

```diff
@@ -9438,6 +9438,10 @@ static ICEDiag CheckICE(const Expr* E, const ASTContext &Ctx) {
   case Expr::CXXFoldExprClass:
   case Expr::CoawaitExprClass:
   case Expr::CoyieldExprClass:
+  case Expr::CountBoundsExprClass:
+  case Expr::InteropTypeBoundsAnnotationClass:
+  case Expr::NullaryBoundsExprClass:
+  case Expr::RangeBoundsExprClass:
```

Comment:
nit: these cases are the same as the case for Expr::PositionalParameterExprClass, again, it could be merged so we keep the checked-c pieces together.